pkg/eval/builtin_fn_str_test.go

```diff
@@ -39,6 +39,8 @@ func TestBase(t *testing.T) {
 		That(`base 16 42 233`).Puts("2a", "e9"),
 		That(`base 1 1`).Throws(ErrBadBase),
 		That(`base 37 10`).Throws(ErrBadBase),
+		That(`base 16 18446744073709551607`).Puts("fffffffffffffff7"),
```

Comment:
The unit tests also need at least one case that explicitly tests passing a num object. Such as

```
That(`base (num 16) (math:pow 2 8)`).Puts( "100"),
```

Reply:
I should have been clearer that my prior comment was not about your change. A test of the sort I proposed should already be present in the project. Had it been present it would have alerted you to the fact your change is incorrect.

Reply:
And, obviously, there should be tests added that explicitly test non-integer values throw the expected exception. Which should already be present but should be added if you want to pursue this change. Such as something like the following:

```
That(`base 8 (num 1/8)`).Throws(something),
```

That's because the current implementation implicitly detects that problem and throws a "wrong type" exception since `num 1/8` is not an integer. But your change requires an explicit test for that scenario since your modified change (per my previous comments) would otherwise accept at `num 1/8` object.

Reply:
Thanks for the comments. I've added more test cases in the new version.